### PR TITLE
restaurant independent project

### PR DESCRIPTION
**Pull request by ninesky00**

### CODE REVIEW

**Comment by KathleenScriver** [comment]
Moving forward, you'll want to commit after every test is written and then after each method is completed.

File: lib/restaurant.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+class Restaurant
+    attr_reader :name, :dishes
+    def initialize(opening_time = "11:30", name)
```
Comment (KathleenScriver) [inline lib/restaurant.rb:3]: No need for a default value here. I do realize that the interaction pattern had an oddity to it though.

File: lib/restaurant.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        @opening_time = opening_time
+        @name = name 
+        @dishes =[]
+    end
+    # would love to refractor how to break up the time input string so code doesn't look like below
+    def opening_time
+        opening_time = Time.new(2020, 10, 07, @opening_time[0..1], @opening_time[3..4])
+        opening_time.strftime("%k:%M")
```
Comment (KathleenScriver) [inline lib/restaurant.rb:11]: I appreciate that you went for a Time object here. Could have also just worked with it as a string too.

File: lib/restaurant.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        closing_time = Time.new(2020, 10, 07, @opening_time) + hours*60*60
+        closing_time.strftime("%k:%M")
+    end
+
+    def add_dish(dish)
+        @dishes << dish
+    end
+
+    def open_for_lunch?
+        if Time.new(2020, 10, 07, @opening_time[0..1], @opening_time[3..4]) < Time.new(2020, 10, 07, 12)
```
Comment (KathleenScriver) [inline lib/restaurant.rb:24]: Some helper methods would have worked well to clean up this if condition. Also, is the if/else/end even necessary? :)

File: lib/restaurant.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            false
+        end
+    end
+
+    def menu_dish_names
+        menu_dish_names =[]
+        @dishes.each do |dish|
+            menu_dish_names << dish.upcase
+        end
+        menu_dish_names
```
Comment (KathleenScriver) [inline lib/restaurant.rb:36]: Great job using .each

File: test/restaurant_test.rb [hunk 5]
```diff
@@ -51,4 +51,30 @@ def test_it_can_add_dishes
...
     assert_equal ['Burrata', 'Pizzetta', 'Ravioli'], restaurant.dishes
   end
+
+  def test_it_is_open_for_lunch
+    #skip
+    restaurant1 = Restaurant.new('10:00', 'Fuel Cafe')
+    restaurant2 = Restaurant.new('16:00', 'Il Poggio')
+
+    assert restaurant1.open_for_lunch?
+    refute restaurant2.open_for_lunch?
```
Comment (KathleenScriver) [inline test/restaurant_test.rb:61]: Good job testing for both the true and false cases.

**Review by KathleenScriver — COMMENTED** [review]
Overall, great job. The Time object was not necessary and probably made your work a lot more difficult. :) Great work!

**Comment by ninesky00** [comment]
Understood, thank you for the corrections. Will definitely remember to commit after every test and method. Will try to clean up my code the best I can.

File: lib/restaurant.rb (hunk 3 above)
Comment (ninesky00) [reply lib/restaurant.rb:24]: totally got trigger happy with the if statement, not needed at all =)

**Pull request closed without merging**